Automattic/wp-calypso
# Site Migration: Remove feature flag  `enable-migration-assistant`

## Proposed Changes


* Remove all references to  `migration-flow/enable-migration-assistant`  form `/setup/migration` and  `/setup/hosted-site-migration` (same of site-migration)
* Remove all references to the old `migration-source-url` step, which only captured the user domain. Now, we can capture way more data with the credentials step. 
* Delete the test cases the feature is off


## Why are these changes being made?
* Code cleanup to improve maintenance

## Testing Instructions
- Go to any of our migration flows (e.g./setup/hosted) 
- Follow the steps and check if you can see the `SITE_MIGRATION_HOW_TO_MIGRATE` step without having the removed feature flag present on this branch. 

## Pre-merge Checklist
